docs/docs/how-to/plugins-and-themes/creating-a-source-plugin.md

```diff
@@ -806,6 +806,183 @@ One challenge when developing locally is that a developer might make modificatio
+
+```javascript:title=source-plugin/utils.js
+export function createNodeManifest({
+  // ...
+}) {
+  // ...
+
+  if (shouldCreateNodeManifest) {
+    // highlight-start
+    const updatedAt = entryItem.updatedAt
+    const manifestId = `${project.id}-${entryItem.id}-${updatedAt}`
+
+    unstable_createNodeManifest({
+      manifestId,
+      node: entryNode,
+      updatedAtUTC: updatedAt,
+    })
+    // highlight-end
+  }
+}
+```
+
+##### Warn if no support
+
+Lastly we'll want to give our users a good experience and give a warning if they're using a version of Gatsby that does not support Content Sync
+
+```javascript:title=source-plugin/utils.js
+// highlight-start
+let warnOnceForNoSupport = false
+// highlight-end
+
+export function createNodeManifest({
+  // ...
+}) {
+  // ...
+
+  if (shouldCreateNodeManifest) {
+    // ...
+  } else if (
+    // highlight-start
+    // it's helpful to let users know if they're using an outdated Gatsby version so they'll upgrade for the best experience
+    isPreview && !createNodeManifestIsSupported && !warnOnceForNoSupport
+  ) {
+    console.warn(
+      `${sourcePluginName}: Your version of Gatsby core doesn't support Content Sync (via the unstable_createNodeManifest action). Please upgrade to the latest version to use Content Sync in your site.`
+    )
+    // This is getting called for every entry node so we don't want the console logs to get cluttered
+    warnOnceForNoSupport = true
+    // highlight-end
+  }
+}
+```
+
+#### Content Sync data source (or CMS) changes
+
+The CMS will need to send a preview webhook to Gatsby Cloud when content is changed and open the Content Sync waiting room. Follow along to learn how to implement it on the CMS side.
+
+##### Using the Gatsby preview extension
+
+We are currently in process of creating an open source package to handle most of this functionality for you. In the meantime, you will need to create a button in your CMS that does the following...
+
+1. `POST` to the preview webhook url in Gatsby Cloud
+2. Open the Content Sync waiting room
+
+The button might look something like this:
+![Open Preview Button Screenshot](https://user-images.githubusercontent.com/18426780/61498855-37e35a00-a982-11e9-8201-ab96be74f1f1.png)
+
+##### Configuration
+
+You will need to store the Content Sync URL from a given Gatsby Cloud site. This will look something like `https://gatsbyjs.com/content-sync/<siteId>`. This is often done in the CMS plugin extension configuration, this will differ from CMS to CMS depending on how they handle their plugin ecosystem.
```

Comment:
We should mention that this needs to be user-configurable. So stored in a CMS options page or something like that.